infra/aws/terraform/prow-build-cluster/iam.tf

```diff
@@ -14,8 +14,87 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+# TODO(pkprzekwas): remove after applying changes on prow-build-cluster
 module "iam" {
+  count = var.cluster_name == "prow-build-cluster" ? 1 : 0
+
   source = "./modules/iam"
 
-  eks_admins = var.eks_admins
+  eks_admins = var.eks_cluster_admins
+}
+
+# We allow Prow Pods with specific service acccounts on the a particular cluster to assume this role.
+resource "aws_iam_role" "eks_admin" {
```

Comment:
Nit: we could rename this Terraform resource to `prow_cluster_admin` to avoid confusion between `eks_admin` and `eks_cluster_admin`